source/development/coredump.rst

```diff
@@ -0,0 +1,160 @@
+        # ulimit -c unlimited
+
+#. **Enable Core Dump Generation**:
+    Enable core dump generation by setting the ``core_pattern`` sysctl parameter to specify the core dump file pattern and location. For example, to set the core dump file pattern to ``/var/core/core.%e.%p`` (where `%e` represents the executable name and `%p` represents the process ID):
+
+    .. code-block:: console
+
+        # echo "/var/core/core.%e.%p" > /proc/sys/kernel/core_pattern
+
+#. **Automate Configuration (Optional)**:
+    To preserve these changes across reboots, add the ``ulimit`` command and ``echo`` command setting ``core_pattern`` to a startup script or system initialization script (e.g., ``/etc/rc.local``).
+
+    By following these steps, you can set up core dump generation manually without relying on systemd. However, keep in mind that the process may vary slightly depending on the Linux distribution and version you are using.
+
+
+To disable core dump generation  you can directly adjust system-wide settings and configurations. Here's how you can do it:
+
+#. **Identify the Target Directory for Core Dumps**:
+    If core dumps are currently being generated, identify the directory where they are stored. By default, core dumps may be stored in the current working directory or in the directory specified by the ``core_pattern`` sysctl parameter.
+
+#. **Disable Core Dump Generation**:
+    To disable core dump generation, set the core dump size limit to zero using the ``ulimit`` command:
+
+    .. code-block:: console
+
+        # ulimit -c 0
+
+#. **Optional: Configure Core Dump Storage Location**:
+    If core dumps were previously being stored, you may want to configure the ``core_pattern`` sysctl parameter to prevent any future core dumps from being generated. For example, you can set it to ``/dev/null`` to discard core dumps:
+
+    .. code-block:: console
+
+        # echo "/dev/null" > /proc/sys/kernel/core_pattern
+
+#. **Restart Processes if Necessary**:
+    If you've changed the ``core_pattern`` parameter, consider restarting relevant processes to ensure that the changes take effect.
+
+By following these steps, you can disable core dump generation without relying on systemd. This approach directly modifies system-wide settings to prevent core dumps from being generated.
+
+MacOS agent's OS
+----------------
+On macOS, core dump generation is disabled by default for most applications. However, you can enable core dump generation for specific processes using the ``ulimit`` command. Here's how you can enable core dump generation on macOS:
+
+#. **Check Current Core Dump Configuration**:
+    Before enabling core dump generation, check the current core dump size limit using the ``ulimit`` command:
+
+    .. code-block:: console
+
+        # ulimit -c
+        # sysctl kern.corefile
+
+#. **Identify the Target Directory for Core Dumps**:
+    On macOS, core dump files are typically stored in the current working directory of the process that crashes.
+
```

Comment:
Why running the `sysctl kern.corefile` command in the previous step? And why a step to identify a target directory for core dumps if we don't provide a command or a way to achieve it?

Reply:
See `sysctl -w kern.corefile=/cores/core.%P` comment below

Reply:
The first `sysctl kern.corefile` command shows the current setting, the second `sysctl`command sets it.

Then I will eliminate this previous control, in the end the setting is better and safer